libs/core/include/CutFlowManager.hpp

```diff
@@ -14,21 +14,19 @@ class CutFlowManager {
   CutFlowManager(std::shared_ptr<EventReader> eventReader_, std::shared_ptr<EventWriter> eventWriter_ = nullptr, std::vector<std::string> additionalCutFlowCollections = {});
   ~CutFlowManager();
 
-  void RegisterCut(std::string cutName);
-  void UpdateCutFlow(std::string cutName);
-  bool HasCut(std::string cutName);
-  void SaveCutFlow();
-  std::map<std::string, float> GetCutFlow();
-  void Print();
-
-  bool isEmpty() { return weightsAfterCuts.empty(); }
-
-  void RegisterCutForCollection(std::string collectionName, std::string cutName);
-  void UpdateCutFlowForCollection(std::string collectionName, std::string cutName);
-  bool HasCutInCollection(std::string collectionName, std::string cutName);
-  void SaveCutFlowForCollections(std::string collectionName);
-  std::map<std::string, float> GetCutFlowForCollection(std::string collectionName);
-  void PrintCutFlowForCollection(std::string collectionName);
+  void RegisterPreExistingCuts(std::string collectionName = "");
+  void RegisterCollection(std::string collectionName);
+
+  void RegisterCut(std::string cutName, std::string collectionName = "");
+  void UpdateCutFlow(std::string cutName, std::string collectionName = "");
+  bool HasCut(std::string cutName, std::string collectionName = "");
+  void SaveCutFlow(std::string collectionName = "");
```

Comment:
Should this perhaps become a private method? And then `SaveAllCutFlows` could perhaps become simply `Save` or `Write`?

Reply:
These cut flows probably take very little space, so I would vote for having just one method that saves all cut flows for simplicity.

Reply:
Okey done